### PR TITLE
New 4/2/1 control event handler

### LINKED ISSUE #3259: new event handler for 4/2/1 control

### PR DESCRIPTION
**Pull request by ruggi**
Fixes #3259 

This PR introduces a new logic to handle submit values from the 4/2/1 control that is more clear than the legacy one _and_ fixes aggregate controls creating multiple undo steps.

### CODE REVIEW

File: editor/src/components/inspector/common/inspector-end-to-end-tests.spec.browser2.tsx [hunk 1]
```diff
@@ -2100,9 +2100,9 @@ describe('inspector tests with real metadata', () => {
           await setControlValue('padding-V', '20', renderResult.renderedDOM)
         },
         endSnippet: makeCodeSnippetWithKeyValue({
-          paddingLeft: 10,
-          paddingTop: 20,
-          paddingBottom: 20,
+          paddingLeft: 10, // no unit because it wasn't affected by the change
+          paddingTop: '20px',
```
Comment (balazsbajorics) [inline editor/src/components/inspector/common/inspector-end-to-end-tests.spec.browser2.tsx:2104]: hmm, the standard Utopia behavior for single values is to omit the `px` when it is not needed. so these tests should remain `paddingTop 20,` I believe

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/inspector/sections/layout-section/layout-system-subsection/split-chained-number-input.tsx [hunk 2]
```diff
@@ -146,44 +152,160 @@ function getInitialMode(
...
-        },
-        transient,
+  (useShorthand: boolean, aggregates: SplitControlValues): void => {
+    const setProp = (path: PropertyPath, values: (CSSNumber | null)[]): EditorAction => {
+      return setProp_UNSAFE(
+        element,
+        path,
+        jsxAttributeValue(
+          mapDropNulls((v) => v, values)
+            .map((v) => printCSSNumber(v, null))
```
Comment (balazsbajorics) [inline editor/src/components/inspector/sections/layout-section/layout-system-subsection/split-chained-number-input.tsx:198]: this is where you want to replace `null` with `'px'` to omit printing the `'px'`  from simple values such as `'20px'`!

**Review by balazsbajorics — APPROVED** [review]
I left a comment about not printing 'px' values when not needed, otherwise looks good to merge!

File: editor/src/components/inspector/sections/layout-section/layout-system-subsection/split-chained-number-input.tsx (hunk 2 above)
Comment (ruggi) [reply editor/src/components/inspector/sections/layout-section/layout-system-subsection/split-chained-number-input.tsx:198]: oooooh nice! thank you 🙏 🙏

File: editor/src/components/inspector/common/inspector-end-to-end-tests.spec.browser2.tsx (hunk 1 above)
Comment (ruggi) [reply editor/src/components/inspector/common/inspector-end-to-end-tests.spec.browser2.tsx]: https://github.com/concrete-utopia/utopia/pull/3260/commits/279a6c1a6993456e01c55101c77d8df32a9ddfaa

**Pull request merged**